### DISCUSSION TITLE
Auto-next question

### DISCUSSION THREAD

**Discussion by ShaineRosewel** [Devographics/surveys#182] [General Survey Design]
**This idea came from the feedbacks received in 2022 State of CSS.** 

Is it possible for us to add a feature that automatically presents the next question on screen so that users won't have to navigate themselves? This can be used with a progress bar so respondents know if it is going to end soon.

I saw something similar in [State of PostgreSQL](https://www.timescale.com/state-of-postgres/2022) - this is open til end of August so you can check it out.

**Advantages:**

- less effort for users to navigate from question to question
- ensures all questions have been viewed (unless terminated early, maybe)

**Disadvantages:**

- could be difficult to go back to a specific question, when needed

Please feel free to add any details to make this more concrete, if this sounds feasible!

@SachaG

**Comment by SachaG** [comment]
It's possible and would be a nice feature to have, but not doable for this year's round of surveys sadly.

**Reply by mikequiapos** [reply]
Was that the Step Form UI. It's great for contextual questions.

**Reply by SachaG** [reply]
I think the link above is to a TypeForm survey? We used TypeForm before developing our own platform actually, it has nice UX but limited features (no localization at the time for example).

**Comment by LeaVerou** [comment]
I’ve been wondering about this as well. Beyond implementation effort, one issue with this is that it prevents respondents from adding additional context through the comment feature.

That said, we could perhaps do it for questions that do *not* support freeform comments. However, if #167 is implemented, that doesn't leave enough to be worth the implementation effort.
